images/wordpress/Dockerfile

```diff
@@ -1,25 +1,24 @@
 FROM php:apache
 
-# APT
 RUN apt-get update && apt-get install -y --no-install-recommends libpng12-dev vim-tiny less mysql-client && rm -r /var/lib/apt/lists/*
 
-# PHP/Apache setup
 RUN docker-php-ext-install mysqli && \
     docker-php-ext-install gd && \
     a2enmod rewrite
 
-# wp-cli
-RUN curl --silent https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar > /usr/local/bin/wp-cli.phar &&\
-    /bin/echo -e '#!/bin/sh\nexec php /usr/local/bin/wp-cli.phar --allow-root "${@}"' > /usr/local/bin/wp &&\
-    chmod 755 /usr/local/bin/wp
+RUN curl --silent https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar > /usr/local/bin/wp-cli.phar
 
-# WordPress
-RUN curl --silent https://wordpress.org/latest.tar.gz > /usr/src/latest.tar.gz && \
-    tar -C /usr/src -xzf /usr/src/latest.tar.gz && \
-    rm -rf /var/www/html /usr/src/wordpress/wp-content && \
-    mv /usr/src/wordpress /var/www/html
+ENV WORDPRESS_VERSION=latest
+
+COPY wp /usr/local/bin/
+COPY wp-start /usr/local/bin/
+
+RUN wp core download --path=/var/www/html
```

Comment:
`wp-start` only downloads WordPress if there's a newer version available (or if the user has changed the WORDPRESS_VERSION variable).

If we removed this line, `wp-start` would need to download a new copy every time.

I just did a very quick, unscientific test. Doing `docker-compose up` takes about 6 seconds in the default configuration (where it just checks for updates), and about 26 seconds when specifying a different WP version.